### PR TITLE
MHR API fix legacy serial number compressed key generation.

### PR DESCRIPTION
**Pull request by doug-lovett**
*Issue #:* /bcgov/entity#17760

*Description of changes:*
- Fix the compressed key generated by the API so it will work with legacy and mhr serial number searches.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by cameron-eyds — APPROVED** [review]
Looks great!

File: mhr_api/src/mhr_api/models/db2/cmpserno.py [hunk 1]
```diff
@@ -12,12 +12,23 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """This module holds data for legacy DB2 compressed serial number information."""
+import re
```
Comment (cameron-eyds) [inline mhr_api/src/mhr_api/models/db2/cmpserno.py:15]: Is this a typo? haha short module name!

Comment (dimak1) [reply mhr_api/src/mhr_api/models/db2/cmpserno.py:15]: lol, I guess it is a short name module `key = re.sub('[^0-9A-Z]+', '', key)`

**Review by dimak1 — APPROVED** [review]

**Comment by RuoxuanPengBC** [comment]
great job, and sorry for my slow code-reading

**Review by RuoxuanPengBC — COMMENTED** [review]
Great job, sorry for my slow code-reading

**Pull request merged**